pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -755,33 +756,56 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 		return statusUnchanged, errors.Errorf("Could not find job %s", s.flinkController.GetLatestJobID(ctx, app))
 	}
 
-	// wait until all vertices have been scheduled and started
-	allVerticesStarted := true
-	for _, v := range job.Vertices {
-		allVerticesStarted = allVerticesStarted && (v.StartTime > 0)
-	}
-
-	if job.State == client.Running && allVerticesStarted {
-		// Update job status
-		jobStatus := s.flinkController.GetLatestJobStatus(ctx, app)
-		jobStatus.JarName = app.Spec.JarName
-		jobStatus.Parallelism = app.Spec.Parallelism
-		jobStatus.EntryClass = app.Spec.EntryClass
-		jobStatus.ProgramArgs = app.Spec.ProgramArgs
-		jobStatus.AllowNonRestoredState = app.Spec.AllowNonRestoredState
-		s.flinkController.UpdateLatestJobStatus(ctx, app, jobStatus)
-		// Update the application status with the running job info
-		app.Status.SavepointPath = ""
-		app.Status.SavepointTriggerID = ""
-		if v1beta1.IsBlueGreenDeploymentMode(app.Status.DeploymentMode) && app.Status.DeployHash != "" {
-			s.updateApplicationPhase(app, v1beta1.FlinkApplicationDualRunning)
+	// wait until all vertices have been scheduled and running
+	jobStartTimeSec := job.StartTime / 1000
+	jobStartTimeNSec := job.StartTime % 1000
+	jobStartTime := time.Unix(jobStartTimeSec, jobStartTimeNSec).UTC()
+	now := time.Now().UTC()
+	if now.Before(jobStartTime.Add(jobVertexStateTimeoutInMinute * time.Minute)) {
+		allVerticesRunning := true
+		hasFailure := false
+		failedVertexIndex := -1
+		for index, v := range job.Vertices {
+			if v.Status == client.Failed || v.Status == client.Failing {
+				failedVertexIndex = index
+				hasFailure = true
```

Comment:
Is this a non recoverable failure?

Reply:
I don't have too much exp about this. Could some vertices fail and succeed after retry?

Reply:
Yeah. I believe so. Real question is what are you addressing here with hasFailure and deployFailed below.